homeassistant/components/demo/camera.py

```diff
@@ -28,16 +35,12 @@ def __init__(self, name):
         self.is_streaming = True
         self._images_index = 0
 
-    def camera_image(self):
+    async def async_camera_image(self):
         """Return a faked still image response."""
         self._images_index = (self._images_index + 1) % 4
+        image_path = Path(__file__).parent / f"demo_{self._images_index}.jpg"
 
-        image_path = os.path.join(
-            os.path.dirname(__file__), f"demo_{self._images_index}.jpg"
-        )
-        _LOGGER.debug("Loading camera_image: %s", image_path)
-        with open(image_path, "rb") as file:
-            return file.read()
+        return await self.hass.async_add_executor_job(_load_image, str(image_path))
```

Comment:
```suggestion
        return await self.hass.async_add_executor_job(_load_image, image_path.read_bytes)
```

Reply:
I thought about that, but didn't do it since there's a test that depends on mocking open and checking string arguments. I didn't want to change the test and the code under test at the same time.

But I can clean it up if we like to.